src/js/ctrl/signals/defaults.ts

```diff
@@ -1,87 +1,98 @@
-'use strict';
-var dl = require('datalib'),
-    ns = require('../../util/ns'),
-    signals = {};
+// TODO(jzong): this probably should be moved to store/factory
 
-var SELECTED = ns('selected'),
-    MODE = ns('mode'),
-    ANCHOR = ns('anchor'),
-    DELTA = ns('delta'),
-    CURSOR = 'cursor',  // Special vega signal, don't namespace.
-    CELL = ns('cell'),
-    MOUSE = ns('mouse');
+import {Map, Record, RecordOf} from 'immutable';
+import {Signal as VegaSignal} from 'vega-typings/types';
+import ns from '../../util/ns';
 
-signals[SELECTED] = {
-  name: SELECTED,
-  init: {mark: {}},
-  streams: [
-    {type: 'mousedown[eventItem().mark && eventItem().mark.name &&' +
-        'eventItem().mark.name !== ' + dl.str(CELL) + ']',
-      expr: 'eventItem()'},
-    {type: 'mousedown[!eventItem().mark]', expr: '{mark: {}}'}
-  ],
-  _idx: 0
-};
-
-signals[MODE] = {
-  name: MODE,
-  init: 'handles',
-  _idx: 1
-};
+const dl = require('datalib');
 
-signals[DELTA] = {
-  name: DELTA,
-  init: 0,
-  streams: [
-    {type: '[mousedown, window:mouseup] > window:mousemove',
-      expr: '{x: eventX() - lyra_anchor.x, y: eventY() - lyra_anchor.y}'}
-  ],
-  _idx: 2
-};
+const SELECTED = ns('selected');
+const MODE = ns('mode');
+const ANCHOR = ns('anchor');
+const DELTA = ns('delta');
+const CURSOR = 'cursor';  // Special vega signal, don't namespace.
+const CELL = ns('cell');
+const MOUSE = ns('mouse');
 
-signals[ANCHOR] = {
-  name: ANCHOR,
-  init: 0,
-  streams: [
-    {type: 'mousedown',
-      expr: '{x: eventX(), y: eventY(), target: eventItem()}'},
-    {type: '[mousedown, window:mouseup] > window:mousemove',
-      expr: '{x: eventX(), y: eventY(), target: lyra_anchor.target}'}
-  ],
-  _idx: 3
-};
+export type LyraSignal = {
+  _idx: number
+} & VegaSignal;
+export const Signal = Record<LyraSignal>({
+  _idx: null,
+  name: null
+});
 
-signals[CELL] = {
-  name: CELL,
-  init: {},
-  streams: [
-    {type: '@' + CELL + ':dragover', expr: 'eventItem()'},
-    {type: '@' + CELL + ':dragleave', expr: '{}'},
-    // {type: '@'+CELL+':mouseover', expr: 'eventItem()'},
-    // {type: '@'+CELL+':mouseout',  expr: '{}'}
-  ],
-  _idx: 4
-};
+export type SignalRecord = RecordOf<LyraSignal>;
 
-signals[MOUSE] = {
-  name: MOUSE,
-  init: {},
-  streams: [
-    {type: 'mousemove, dragover', expr: '{x: eventX(), y: eventY()}'}
-  ],
-  _idx: 5
-};
+export type SignalState = Map<string, SignalRecord>;
 
-signals[CURSOR] = {
-  name: CURSOR,
-  streams: [
-    {type: 'mousedown', expr: "eventItem() && eventItem().cursor || 'default'"},
-    {type: 'mouseup', expr: "'default'"}
-  ]
-};
+export const defaultSignalState: SignalState = Map({
+  SELECTED: Signal({
+    name: SELECTED,
+    value: {mark: {}},
+    on: [
+      {events: 'mousedown[eventItem().mark && eventItem().mark.name &&' +
```

Comment:
These should be updated to the new [event functions](https://vega.github.io/vega/docs/expressions/#event-functions). For example, it's just `item()` now instead of `eventItem()`.

Reply:
done in https://github.com/vega/lyra/pull/427/commits/5b6ce971270944c0ea2e7dd3eb59e089c46c55b6